todo_app/app.py

```diff
@@ -1,11 +1,24 @@
-from flask import Flask
 
+from flask import Flask, render_template, request, redirect
+from todo_app.data.session_items import get_items
+from todo_app.data import session_items
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
-
 @app.route('/')
 def index():
-    return 'Hello World!'
+    todo_items = session_items.get_items()
+    return render_template ('index.html',items=todo_items)
+
+@app.route('/add', methods=['GET', 'POST'])
```

Comment:
Should just be `methods=['POST']`

Your form correctly has `action="POST"` so you don't expect anyone to be making GET requests to "/add".